mewarpx/tests/test_emission.py

```diff
@@ -672,3 +670,69 @@ def test_plasma_injector_fixedT2():
             df[label_base + key + '_std'] = np.std(val)
 
     assert testing_util.test_df_vs_ref(testname=name, df=df, margin=0.3)
+
+
+def test_arbitrary_distribution_emitter():
+    name = "arbitrary_distribution_emitter"
+    # Include a random run number to allow parallel runs to not collide.  Using
+    # python randint prevents collisions due to numpy rseed below
+    testing_util.initialize_testingdir(name)
+
+    # Initialize each run with consistent, randomly-chosen, rseed. Use a random
+    # seed instead for initial dataframe generation.
+    # np.random.seed(11874889)
+
+    run = diode_setup.DiodeRun_V1(
+        GEOM_STR='XZ', PERIOD=0.8e-06 * 512, D_CA=0.8e-06 * 512, DT=1e-12,
+        TOTAL_TIMESTEPS=1, DIAG_STEPS=1
+    )
+
+    run.setup_run(init_inert_gas=True, init_scraper=False, init_injectors=False)
+
+    z = np.arange(0, 512) + 0.5
+    sin_func = np.pi/2 * 1e13 * np.sin(np.pi / 512 * z)
```

Comment:
I would rather have a function that varies both with x and z. If it's too noisy to resolve every cell of the 2D simulation against the analytic function, you could look at both an average of x and an average of z compared to the analytic function or something.